### DISCUSSION TITLE
Changing the ctor for LevelObject to reflect BarricadeManager.spawnBarricade

### DISCUSSION THREAD

**Discussion by Jdance-Media** [SmartlyDressedGames/Unturned-3.x-Community#4053] [Ideas]
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/67397422/cdc7b584-5191-4ce3-9188-ba9ab75e431b)
Using `Transform.GetOrAddComponent` could really help us out when it comes to advanced modding. Also allows us to implement #4051 without you having to add yet another `Interactable` type. It seems you expected us to do something like this when you did the same for `BarricadeManager.spawnBarricade`.

Further explanation:

A little while back I made a money printer that would automatically harvest itself after a timer modhook was triggered. From this, I figured that by adding an `Interactable` component to my main prefab, it would automatically become the one used by the game. This was true, but only in the case of barricades, because at some point you made it so `BarricadeManager.spawnBarricade` used `GetOrAddComponent` rather than `AddComponent`. I attempted to apply the same concept with others using objects, in this case I was using the `InteractableObjectQuest` component, this of course did not work because objects weren't setup the same way and didn't 'adopt' the component I had already attached to the prefab.

Gamingtoday093 pointed out `BarricadeManager.spawnBarricade`. DrDallas and Loco helped test some of theories.

**Comment by SDGNelson** [comment]
Sorry, the barricade one is unintentional and I'd be hesitant to make this change because those components aren't expected to be already existing on any objects.
